src/components/ProjectForm.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Button, TextInput, TextArea } from '../components/Input'
+import { H1, P, Label } from './Typography'
+
+const TableForm = styled.form`
+  display: table;
+  position: relative;
+
+  & > div {
+    display: table-row;
+
+    & > * {
+      display: table-cell;
+    }
+
+    & > label {
+      vertical-align: middle;
+    }
+  }
+`
+
+const RightButton = styled(Button)`
+  float: right;
+`
+
+const SubmitRow = ({ project, submitCallback }) => {
+  return (
+    <div>
+      <div></div>
+      <RightButton
+        onClick={e => submitCallback(project)}
+        type="submit"
+        width="flex"
+        color="primary"
+      >
+        Update Submission
+      </RightButton>
+    </div>
+  )
+}
+
+const FormRow = ({ id, label, children }) => (
+  <div>
+    <Label htmlFor={id}>{label}</Label>
+    <div id={id}>{children}</div>
+  </div>
+)
+
+export default ({ project, submitCallback, onChange }) => {
+  return project ? (
+    <div>
+      <H1>Edit "{project.title}"</H1>
+      <TableForm>
+        <FormRow id="title" label="Project Name">
+          <TextInput
+            value={project.title}
+            placeholder="Required"
+            onChange={e => onChange({ ...project, title: e.target.value })}
```

Comment:
I wonder if passing entire project objects on every keypress could be slow on bad computers. But the code sure is clean... Don't think it needs to be changed but

Reply:
project object isn't that large so I think it should be fine?